Eth eip1559

Tomasvrba opened the pull request.

benma commented on src/eth.rs:
```diff
@@ -444,6 +485,52 @@ impl<R: Runtime> PairedBitBox<R> {
+    /// `raw_tx_slice.try_into()` (`rlp` feature required).
+    pub async fn eth_sign_1559_transaction(
+        &self,
+        chain_id: u64,
```
no need for this argument - it is already in the `tx` struct

benma commented on src/eth.rs:
```diff
@@ -101,6 +101,27 @@ pub struct Transaction {
+    serde(rename_all = "camelCase")
+)]
+pub struct EIP1559Transaction {
+    pub decoded_chain_id: Vec<u8>,
```
why not `chain_id: u64`? it is `u64` pretty much everywhere I think

Tomasvrba replied:
Because the current implementation of the rlp decoding returns all fields as Vec<u8>:
```
let [decoded_chain_id, nonce, max_priority_fee_per_gas, max_fee_per_gas, gas_limit, recipient, value, data, _, _, _]: [Vec<u8>; 11]
```

I cast it as u64 in eth_sign_1559_transaction

Tomasvrba replied:
yeah, good point about the chainIDs bigger than 1 byte, duh 🤦‍♂️ 
added the sandbox signing and fixed here (tested with large chainIDs): https://github.com/digitalbitbox/bitbox-api-rs/pull/59/commits/6ad12788896325c90740a00801186c9410617d92

benma commented on src/eth.rs:
```diff
@@ -145,10 +145,14 @@ impl TryFrom<&[u8]> for EIP1559Transaction {
+        while chain_id.len() < 8 {
+            chain_id.insert(0, 0);
+        }
+        chain_id = u64::from_be_bytes(chain_id.try_into().expect("chain_id must be 8 bytes long after padding"));
```
no need to panic here, you can just return an error `...try_into().map_err(|_| ())?`.

benma commented:
@Tomasvrba seems like you should setup rustfmt in your editor to format your code on save :)

benma commented:
@Tomasvrba there are more issues with rustfmt and also with test/lint.

Run `./ci.sh` locally to see one issue, then repeat until there are no more issues :)

Tomasvrba commented:
> @Tomasvrba there are more issues with rustfmt and also with test/lint.
> 
> Run `./ci.sh` locally to see one issue, then repeat until there are no more issues :)

ok, my local ci now passes, can you run it here (looks like I can't start it)
squashed previous changes and just added the most recent one as the new !fixup commit

Tomasvrba commented:
> please revert the Cargo.lock changes, I don't think updating deps has anything to do with this change, right?
> 
> otherwise tACK, working well!

ok, squashed and removed

benma commented:
Thanks a lot! :heart:
